File: experiments/libs/server/dotnet/DummyExperimentServer.cs
using System;
using System.IO;

class SampleExperimentServer : WebLabDeusto.ExperimentServer {
        public string SendFile(byte [] file, string fileInfo){
            int length = file.Length;
            Console.WriteLine("File received: {0}", length);
            return "File received " + length;
        }

        public string SendCommand(string command){
            Console.WriteLine("Command received: {0}", command);
            return "Command received: " + command;
        }

        public void StartExperiment(){
            Console.WriteLine("Experiment started");
        }

        public void Dispose(){
            Console.WriteLine("Experiment disposed");
        }
}

class Tester{
    public static void Main(){
        WebLabDeusto.Runner runner = new WebLabDeusto.Runner(
                                            new SampleExperimentServer(),
                                            5678,
                                            "weblab"
                                    );
        runner.Start();
        Console.WriteLine("Press to shutdown"); 
        Console.ReadLine(); 
    
    }
}

File: experiments/libs/server/java/WebLabJavaXmlRpcServer/src/com/example/weblab/DummyExperimentServer.java
package com.example.weblab;

import java.io.File;

import es.deusto.weblab.experimentservers.IExperimentServer;
import es.deusto.weblab.experimentservers.WebLabException;

public class DummyExperimentServer implements IExperimentServer {
	
	public String sendCommand(String command)  throws WebLabException {
		System.out.println("I'm at send_command: " + command);
		return "ok";
	}

	public String sendFile(File file, String fileInfo)  throws WebLabException {
		System.out.println("I'm at send_program: " + file.getAbsolutePath() + "; fileInfo: " + fileInfo);
		return "ok";
	}
	
	public void dispose() {
		System.out.println("I'm at dispose");
	}

	public void startExperiment() throws WebLabException {
		System.out.println("I'm at startExperiment");
	}
}
